nnsmith/materialize/tensorflow/forward.py

```diff
@@ -293,3 +293,86 @@ def forward_fn(op: NHWCConv2d):
         dtype=op.input_like[0].dtype.tensorflow(),
         autocast=False,
     )
+
+
+@operator_impl(Squeeze)
+def forward_fn(op: Squeeze):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.squeeze(x, axis=op.extra_attrs["reduce_dim"])
+    else:
+        return lambda x: tf.squeeze(x)
+
+
+@operator_impl(ReduceSum)
+def forward_fn(op: ReduceSum):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.reduce_sum(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.reduce_sum(x)
+
+
+@operator_impl(ReduceMin)
+def forward_fn(op: ReduceMin):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.reduce_min(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.reduce_min(x)
+
+
+@operator_impl(ReduceMax)
+def forward_fn(op: ReduceMax):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.reduce_max(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.reduce_max(x)
+
+
+@operator_impl(ReduceMean)
+def forward_fn(op: ReduceMean):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.reduce_mean(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.reduce_mean(x)
+
+
+@operator_impl(ReduceProd)
+def forward_fn(op: ReduceProd):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.reduce_prod(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.reduce_prod(x)
+
+
+@operator_impl(ArgMin)
+def forward_fn(op: ArgMin):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.argmin(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.argmin(x)
+
+
+@operator_impl(ArgMin)
+def forward_fn(op: ArgMin):
+    if op.extra_attrs["reduce_dim"] is not None:
+        return lambda x: tf.math.argmax(x, axis=op.extra_attrs["reduce_dim"])
+    return lambda x: tf.math.argmax(x)
+
+
+@operator_impl(Tril)
+def forward_fn(op: Tril):
+    return lambda x: tf.experimental.numpy.tril(x, k=op.diagonal)
+
+
+@operator_impl(Triu)
+def forward_fn(op: Triu):
+    return lambda x: tf.experimental.numpy.triu(x, k=op.diagonal)
+
```

Comment:
I'm not sure if they can directly cooperate with XLA/TFLite/IREE, because these two come from `tf.experimental.numpy` instead of directly coming from some supported operator domains (like `raw_ops`, `math`, etc.). Maybe we want to double-check this a little bit.

Reply:
I think this can be handled by dtype test?

Reply:
Yes. I just want to confirm that those two can work some backends smoothly, otherwise, we don't have to maintain them. :-)

Reply:
I tested them on XLA and TFLite, and they worked fine (though failed with some dtype like bool, but ok for integers and floats).